### PR TITLE
fix: Select benchmarks CLI option

### PR DESCRIPTION
**Pull request by isaac-chung**
Address #1250 point 2.
- Added test cases for a list of Benchmark objects passed into the task parameter
- Added command line option to select Benchmarks

The following can now run. 1) script
```python
from sentence_transformers import SentenceTransformer
import mteb

model_name = "average_word_embeddings_komninos"
model = SentenceTransformer(model_name)
benchmark = mteb.get_benchmarks(names=["MTEB(eng)"])
evaluation = mteb.MTEB(tasks=benchmark)
results = evaluation.run(model)
```
and command line:
```
mteb run -m average_word_embeddings_komninos -b 'MTEB(eng)'
```

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

### CODE REVIEW

File: mteb/evaluation/MTEB.py [hunk 1]
```diff
@@ -58,6 +59,11 @@ def __init__(
 
         if tasks is not None:
             self._tasks = tasks
+            if (
+                not isinstance(tasks[0], AbsTask) and not isinstance(tasks[0], str)
+            ):  # So that passing in a list of Benchmarks works too. Trying to import Benchmark causes circular import.
```
Comment (KennethEnevoldsen) [inline mteb/evaluation/MTEB.py:64]: Hmm, I think we should use Benchmark here, why do we see the circular import? (something we should fix?)

Sidenote: https://github.com/embeddings-benchmark/mteb/issues/1272

**Review by KennethEnevoldsen — APPROVED** [review]
only a few minor things.

File: mteb/evaluation/MTEB.py (hunk 1 above)
Comment (isaac-chung) [reply mteb/evaluation/MTEB.py]: Done. And also commented on the tagged issue.

**Comment by isaac-chung** [comment]
Thanks! 🚀 Enabling auto-merge.

**Pull request merged**